zulipterminal/ui_tools/boxes.py

```diff
@@ -445,11 +445,23 @@ def autocomplete_users(self, text: str, prefix_string: str
                                        reverse=True)
 
         user_names = [user['full_name'] for user in sorted_matching_users]
+
+        user_name_duplicates = [user_name
+                                for count, user_name in enumerate(user_names)
+                                if user_name in user_names[:count]]
+
+        user_names_with_distinct_duplicates = [
+            f"{user['full_name']}|{user['user_id']}"
+            if user['full_name'] in user_name_duplicates
+            else user['full_name']
```

Comment:
Minor: I know you need user as the base object to get the user-id in the loop, but it seems strange that we're generating a list with names and then only use it once.